sympy/matrices/matrices.py

```diff
@@ -4366,7 +4368,11 @@ def norm(self, ord=None):
 
         # Matrix Norms
         else:
-            if ord == 2:  # Spectral Norm
+            if ord == 1:  # Maximum column sum
+                def maxcol(x) : return sum(self.applyfunc(lambda y : abs(y)).col(x))
```

Comment:
`applyfunc` is run repeatedly for each call of `maxcol`. That is not efficient. I would first save a copy of the matrix with absolute values of entries: 

    m = self.applyfunc(abs).

(`lambda y: abs(y)` is the same function as `abs`.)

Moreover, it is not necessary to introduce a new function `maxcol`.  The list of column sums can be obtained directly from the Python generator expression

    sum(m.col(i)) for i in range(m.cols))

There only remains to apply `max` to this result.